File: agora-easycall-ios-oc/AgoraSample/Menu/AGSMenuPushSegue.m
//
//  AGSMenuPushSegue.m
//  AgoraSample
//
//  Created by Shilong on 9/17/15.
//  Copyright (c) 2015 Agora. All rights reserved.
//

#import "AGSMenuPushSegue.h"
#import "AGSFrostedViewController.h"
#import "AGSMenuViewController.h"
#import "AGSChatViewController.h"
#import "AGSAgoraKitManager.h"

@implementation AGSMenuPushSegue

- (void)perform
{
    UIViewController *menuViewController = (AGSMenuViewController *)self.sourceViewController;
    UIViewController *nextViewController = (UIViewController *)self.destinationViewController;
    UINavigationController *currentNavigationController = (UINavigationController *)menuViewController.frostedViewController.contentViewController;    
    
    [menuViewController.frostedViewController hideMenuViewControllerWithCompletionHandler:^{
        
        if ([currentNavigationController.topViewController isKindOfClass:[AGSChatViewController class]]) {
            AGSChatViewController *chatViewController = (AGSChatViewController *)currentNavigationController.topViewController;
            [AGSAgoraKitManager shareManager].callBackView.durationTime = chatViewController.duration;
            [[AGSAgoraKitManager shareManager].callBackView showCallBackView];
        }
        
        [currentNavigationController pushViewController:nextViewController animated:YES];
    }];
}

@end

File: agora-easycall-ios-swift/SampleSwift/AGSMenuPushSegue.swift
//
//  AGSMenuPushSegue.swift
//  SampleSwift
//
//  Created by let on 10/29/15.
//  Copyright © 2015 Agora. All rights reserved.
//

import UIKit

class AGSMenuPushSegue: UIStoryboardSegue {

    override func perform() {
        let menuViewController = sourceViewController as? AGSMenuViewController
        let nextViewController = destinationViewController
        let currentNavigationController = menuViewController?.frostedViewController.contentViewController as? UINavigationController
        
        menuViewController?.frostedViewController.hideMenuViewControllerWithCompletionHandler({ () -> Void in
            if currentNavigationController?.topViewController is AGSChatViewController {
                let chatViewController = currentNavigationController?.topViewController as? AGSChatViewController
                AGSAgoraKitManager.shareManager.callBackView.durationTime = Int((chatViewController?.duration)!)
                AGSAgoraKitManager.shareManager.callBackView.showCallBackView()
            }
            
            currentNavigationController?.pushViewController(nextViewController, animated: true)
        })
    }
}
